### DISCUSSION TITLE
how to update

### DISCUSSION THREAD

**Discussion by usmcsni** [OctoFarm/OctoFarm#409] [Q&A] [answered]
what's the easiest way to update an existing octofarm install on a raspberry pi

**Comment by NotExpectedYet** [comment, marked as answer]
Hello

`cd ~/OctoFarm` - Make sure your in the OctoFarm directory. This command is making the assumption it's installed in your user home directory as per instructions on website. 
`pm2 stop OctoFarm`
`git pull`
`npm ci --production`
`pm2 start OctoFarm` 

Should do it.

**Reply by davidzwa** [reply]
Just a suggestion, maybe `npm ci --production` is a better idea as the OP wouldnt be downloading testing stuff.

**Reply by usmcsni** [reply]
got in did the pm2 stop
but git pull is not working says not a git repository

**Reply by davidzwa** [reply]
Ok so in order to do git pull a `.git` folder needs to be present. Also you need to do this command within the OctoFarm folder where this .git subfolder is present. How did you download OctoFarm?

In order to be able to pull with git, you'd need to clone it (not download a ZIP).
Please follow the instructions on how to clone here https://github.com/OctoFarm/OctoFarm

We have work ready on making this process easier, which will be out soon.

**Reply by gmccauley** [reply]
> but git pull is not working says not a git repository

Make sure you change directory to be in the OctoFarm folder before you run 'git pull'

**Reply by NotExpectedYet** [reply]
Cheers @gmccauley updated my comment.
